app/models/server.rb

```diff
@@ -15,6 +15,10 @@ class Server < ApplicationRecord
   validates :ram_gb, numericality: { greater_than: 0 }
   validates :storage_gb, numericality: { greater_than: 0 }
 
+  MAX_NAME_LENGTH = 20
```

Comment:
Can we add a comment (e.g. a link to the vSphere docs) where this hard limit is documented?

Reply:
I'm actually not sure where this limit originates from, vSphere/Puppetscripts, etc.
As Servers are actually not vSphere objects (or are they?) they should not have this restriction anyway.
In any case, this code is mostly WIP, as its almost entirely copied from VMConfigs, I still have to remove all business logic that does not match servers.
Thank you for commenting anyway, I will look into it when I got the basic user management up and running

Reply:
> As Servers are actually not vSphere objects

Correct.

> I will look into it when I got the basic user management up and running

Thanks!

Reply:
> Can we add a comment (e.g. a link to the vSphere docs) where this hard limit is documented?

I actually could not find any reason why the name should be length-restricted.
Neither the vSphere-Docs, nor the puppet docs mention any such restrictions
I asked @BjoernDaase who previously was a PO, and he says it was a requirement by the customer, but he does not remember why.

I removed the length restriction and lower-case-only restriction from servers, as they are not a technical necessity. I left them in place for VMs though.